src/main/java/org/isf/utils/jobjects/PaginatedTableDecorator.java

```diff
@@ -0,0 +1,372 @@
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program. If not, see <http://www.gnu.org/licenses/>.
+ */
+package org.isf.utils.jobjects;
+
+import java.awt.BorderLayout;
+import java.awt.Component;
+import java.awt.GridLayout;
+import java.awt.Insets;
+import java.awt.event.ActionEvent;
+import java.awt.event.KeyEvent;
+import java.util.Arrays;
+import java.util.List;
+
+import javax.swing.Box;
+import javax.swing.ButtonGroup;
+import javax.swing.JButton;
+import javax.swing.JComboBox;
+import javax.swing.JLabel;
+import javax.swing.JPanel;
+import javax.swing.JScrollPane;
+import javax.swing.JTable;
+import javax.swing.JToggleButton;
+import javax.swing.SwingConstants;
+import javax.swing.event.RowSorterEvent;
+import javax.swing.event.TableModelEvent;
+import javax.swing.table.TableModel;
+
+/**
+ * Idea inspired by <a href="https://www.logicbig.com/tutorials/java-swing/jtable-lazy-pagination.html">LogicBig - JTable Lazy Pagination with JPA</a><br>
+ * <br>
+ * {@link PaginatedTableDecorator} is JTable decorator that shows a page selector at the top of the scrollpane, also giving the ability to choose the page
+ * size.<br>
+ * <br>
+ * It works with:
+ * <ul>
+ * <li>{@link PageableTableModel}: the table model must be of type PageableTableModel<T>
+ * <li>{@link PaginationDataProvider}: the container class has to implement the interface PaginationDataProvider<T>
+ * </ul>
+ * 
+ * It offers three different layout:
+ * <ul>
+ * <li>{@link PAGELINK}: clickable page numbers with auto-layout
+ * <li>{@link PAGEBROWSER}: standard page selector with arrows for first, previous, next and last page
+ * <li>{@link PAGEBROWSER_PAGELINK}: both the previous ones
+ * </ul>
+ * 
+ * @author Nanni
+ * 
+ * @param <T>
+ *            - The model to be paginated
+ */
+public class PaginatedTableDecorator<T> {
+
+	/** PAGELINK: clickable page numbers with auto-layout */
+	public static final int PAGELINK = 0;
+	/** PAGEBROWSER: standard page selector with arrows for first, previous, next and last page */
+	public static final int PAGEBROWSER = 1;
+	/** PAGEBROWSER_PAGELINK: standard page selector with arrows and clickable page numbers with auto-layout */
+	public static final int PAGEBROWSER_PAGELINK = 2;
+	protected int paginatorType;
+
+	private JTable table;
+	private PaginationDataProvider<T> dataProvider;
+	private int[] pageSizes;
+	private JPanel contentPanel;
+	private int currentPageSize;
+	private int currentPage = 1;
+	private int lastPage = 1;
+	private JPanel pageLinkPanel;
+	private PageableTableModel objectTableModel;
+	private JButton jFirstPageButton;
+	private JButton jPreviousPageButton;
+	private JButton jNextPageButton;
+	private JButton jLastPageButton;
+	private JLabel currentPageLabel;
+	private static final int MaxPagingCompToShow = 9;
```

Comment:
Rename the static final to `MAX_PAGING_COMP_TO_SHOW`

Reply:
Done.